lib/instance/pages/instance_page.dart

```diff
@@ -0,0 +1,370 @@
+                                    children: [
+                                      Text(l10n.posts),
+                                    ],
+                                    onPressed: () async {
+                                      viewType = SearchType.posts;
+                                      await context.read<InstancePageCubit>().loadPosts(page: 1);
+                                      WidgetsBinding.instance.addPostFrameCallback((_) => _scrollController.jumpTo(0));
+                                    },
+                                  ),
+                                  const SizedBox(width: 10),
+                                  SearchActionChip(
+                                    backgroundColor: viewType == SearchType.comments ? theme.colorScheme.primaryContainer.withOpacity(0.25) : null,
+                                    children: [
+                                      Text(l10n.comments),
+                                    ],
+                                    onPressed: () async {
+                                      viewType = SearchType.comments;
+                                      await context.read<InstancePageCubit>().loadComments(page: 1);
+                                      WidgetsBinding.instance.addPostFrameCallback((_) => _scrollController.jumpTo(0));
+                                    },
+                                  ),
+                                ],
+                              ),
+                            ),
+                          ),
+                        ),
+                      ),
+                      if (state.status == InstancePageStatus.loading)
+                        const SliverFillRemaining(
+                          child: Center(
+                            child: CircularProgressIndicator(),
+                          ),
+                        ),
+                      if (state.status == InstancePageStatus.failure)
+                        SliverFillRemaining(
+                          child: ErrorMessage(
+                            message: state.errorMessage,
+                            actionText: l10n.refreshContent,
+                            action: () async => await _doLoad(context),
+                          ),
+                        ),
+                      if (state.status == InstancePageStatus.success || state.status == InstancePageStatus.done) ...[
+                        if (viewType == SearchType.all)
+                          SliverToBoxAdapter(
+                            child: Padding(
+                              padding: const EdgeInsets.all(20),
+                              child: Material(
+                                child: InstanceView(site: widget.getSiteResponse.siteView.site),
+                              ),
+                            ),
+                          ),
+                        if (viewType == SearchType.communities)
+                          SliverToBoxAdapter(

```

Comment:
Is it possible to use `SliverList` or some form of a Sliver-based widget to generate the individual entries? (for communities, users, etc.)

The reasoning here is that with large lists, using `ListView.builder` and `shrinkWrap: true` causes performance issues (which is the reason I switched over to using Slivers in the FeedPage refactor). When using `shrinkWrap`, the whole list is kept in memory and Flutter is unable to remove parts of the list that are not visible to the screen. 

This won't be noticeable for small lists, but is more apparent in larger/infinite lists

https://api.flutter.dev/flutter/widgets/ScrollView/shrinkWrap.html

Reply:
I tried changing from
* `SliverToBoxAdapter` -> `Material` -> `ListView.builder` -> `<widget>`

to

* `SliverList` -> `SliverChildBuilderDelegate` -> `Material` -> `<widget>`

and it seems to work fine. Is that what you had in mind?

c6831c68e3325741e4a8831faaed348b69ea43fe

Reply:
Yup, that looks good!